app/services/teacher_training_api/publish_provider_checker.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module TeacherTrainingApi
+  class PublishProviderChecker
+    include ServicePattern
+
+    attr_reader :recruitment_cycle_year, :provider_matches, :lead_school_matches, :lead_partner_matches, :missing
+
+    def initialize(recruitment_cycle_year:)
+      @recruitment_cycle_year = recruitment_cycle_year
+      @provider_matches = []
+      @lead_school_matches = []
+      @lead_partner_matches = []
+      @missing = []
+    end
+
+    def call
+      next_link = publish_provider_endpoint
+      while next_link.present?
+        response = TeacherTrainingApi::Client::Request.get(next_link).parsed_response
+        response["data"].map { |p| p["attributes"] }.each do |provider|
+          if lead_school_matches?(provider)
+            lead_school_matches << provider
+          elsif lead_partner_matches?(provider)
+            lead_partner_matches << provider
+          elsif provider_matches?(provider)
+            provider_matches << provider
+          else
+            missing << provider
+          end
+        end
+        next_link = response["links"]["next"]
+      end
+
+      self
+    end
+
+    def total_count
+      lead_school_matches.count + lead_partner_matches.count + provider_matches.count + missing.count
+    end
+
+  private
```

Comment:
Appease the rubocop gods 🔱 
```suggestion
  private
  
```